utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala

```diff
@@ -102,7 +102,10 @@ object BestEffortJsonSchemaEncoder {
     case ObjectField(fieldName, None, Some(_), parentSchema)
         if parentSchema.getRequiredProperties.contains(fieldName) =>
       error(s"Missing property: $fieldName for schema: $parentSchema.")
-    case ObjectField(fieldName, Some(value), Some(schema), _) =>
+    case ObjectField(fieldName, Some(value), Some(schema), parentSchema)
+        if parentSchema.getRequiredProperties.contains(fieldName) =>
+      encode(value, schema, Some(fieldName)).map(fieldName -> _)
+    case ObjectField(fieldName, Some(value), Some(schema), _) if value != null =>
```

Comment:
maybe we can join this case with the previous one using `||`. The outcome looks similar.